### PR TITLE
Fixes #20642 - don't set hostname-override when localhost

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

File: templates/rhsm-katello-reconfigure.erb [hunk 1]
```diff
@@ -100,8 +100,8 @@ fi
 
 
 FQDN=`hostname -f`
-if [ $? == "0" ] && [ -d /etc/rhsm/facts/ ]; then
-  echo "{\"network.hostname-override\":\"$FQDN\"}" > /etc/rhsm/facts/katello.facts
+if [ $? == "0" ] && [ "$FQDN" != "localhost" ] && [ -d /etc/rhsm/facts/ ]; then
+  echo "{\"network.hostname-override\":\"$FQDN\"}" > /etc/rhsm/faacts/katello.facts
```
Comment (ekohl) [inline templates/rhsm-katello-reconfigure.erb]: I doubt this line was intentional.

Comment (iNecas) [inline templates/rhsm-katello-reconfigure.erb]: Daaamn :) Fixed. Thanks for spotting this. My future that would need to debug the issue in few years also thanks you :)

**Comment by iNecas** [comment]
@ehelms anything else required for this PR, or it can be merged?

**Comment by iNecas** [comment]
Here is the issue against Katello http://projects.theforeman.org/issues/20816

**Pull request merged**